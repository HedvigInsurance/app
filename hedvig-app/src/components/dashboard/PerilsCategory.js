import React from "react"
import {
  View,
  ScrollView,
  Button,
  Text,
  Image,
  TouchableOpacity
} from "react-native"
import Peril from "../../containers/dashboard/Peril"
import styled from "styled-components/native"
import {
  StyledCategoryContainer,
  StyledCategoryHeader,
  StyledCategoryIcon,
  StyledCategoryTextContainer,
  StyledExpandButton,
  StyledPerilsContainer,
  StyledPerilsRow
} from "../styles/dashboard"
import { DisabledCollapseButton, DisabledExpandButton } from "../Button"
import { StyledRow, StyledWrappingRow } from "../styles/general"
import { StyledHeading, StyledPassiveText } from "../styles/text"
const R = require("ramda")

export class PerilsCategory extends React.Component {
  state = {
    showCategory: false
  }

  coveredStates = ["COVERED", "ADD_REQUESTED", "ADD_PENDING"]

  coveredPerils(enableRemove = false) {
    return R.filter(
      peril => this.coveredStates.includes(peril.state),
      this.props.perils
    ).map((peril, i) => {
      let perilIndex = this.props.perils.findIndex(p => p.id === peril.id)
      return (
        <Peril
          peril={peril}
          key={i}
          enableRemove={enableRemove}
          navigation={this.props.navigation}
          categoryPerils={this.props.perils}
          categoryTitle={this.props.title}
          perilIndex={perilIndex}
        />
      )
    })
  }

  uncoveredPerils() {
    return R.filter(
      peril => !this.coveredStates.includes(peril.state),
      this.props.perils
    ).map((peril, i) => {
      return (
        <Peril
          peril={peril}
          key={i}
          enableAdd={true}
          navigation={this.props.navigation}
          categoryPerils={this.props.perils}
          perilIndex={i}
        />
      )
    })
  }

  allPerils() {
    return this.props.perils.map((peril, i) => {
      return (
        <Peril
          peril={peril}
          key={i}
          navigation={this.props.navigation}
          categoryPerils={this.props.perils}
          perilIndex={i}
        />
      )
    })
  }

  perils() {
    if (this.props.editMode) {
      return (
        <StyledPerilsContainer>
          <StyledPerilsRow>{this.coveredPerils(true)}</StyledPerilsRow>
          <StyledPerilsRow>{this.uncoveredPerils()}</StyledPerilsRow>
        </StyledPerilsContainer>
      )
    } else if (this.state.showCategory) {
      return (
        <StyledPerilsContainer>
          <StyledPerilsRow>{this.coveredPerils()}</StyledPerilsRow>
        </StyledPerilsContainer>
      )
    }
  }

  render() {
    return (
      <TouchableOpacity
        onPress={() =>
          this.setState({ showCategory: !this.state.showCategory })}
      >
        <StyledCategoryContainer>
          <StyledCategoryHeader>
            <StyledCategoryIcon source={{ uri: this.props.iconUrl }} />
            <StyledCategoryTextContainer>
              <StyledRow>
                <StyledHeading>{this.props.title}</StyledHeading>
              </StyledRow>
              <StyledRow>
                <StyledPassiveText>
                  {this.state.showCategory
                    ? "Klicka på ikonerna för mer info"
                    : this.props.description}
                </StyledPassiveText>
              </StyledRow>
            </StyledCategoryTextContainer>
            <StyledExpandButton>
              {this.state.showCategory ? (
                <DisabledCollapseButton />
              ) : (
                <DisabledExpandButton />
              )}
            </StyledExpandButton>
          </StyledCategoryHeader>
          {this.perils()}
        </StyledCategoryContainer>
      </TouchableOpacity>
    )
  }
}
