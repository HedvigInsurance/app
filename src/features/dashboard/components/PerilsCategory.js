import React from 'react';
import { View } from 'react-native';
import { Peril } from './Peril';
import {
  StyledCategoryContainer,
  StyledCategoryHeader,
  StyledCategoryIcon,
  StyledCategoryTextContainer,
  StyledExpandButton,
  StyledPerilsContainer,
  StyledPerilsRow,
  StyledCategoryTextAndButton,
  StyledPerilsHelpText,
} from '../styles/dashboard';
import {
  DisabledCollapseButton,
  DisabledExpandButton,
} from '../../../components/Button';
import { StyledRow } from '../../../components/styles/general';
import {
  StyledHeading,
  StyledPassiveText,
  StyledSmallPassiveText,
} from '../../../components/styles/text';

const coveredStates = ['COVERED', 'ADD_REQUESTED', 'ADD_PENDING'];

export const coveredPerils = (perils, title) => {
  return perils
    .filter((peril) => coveredStates.includes(peril.state))
    .map((peril, index) => (
      <Peril
        peril={peril}
        key={index}
        categoryPerils={perils}
        categoryTitle={title}
        perilIndex={perils.findIndex((p) => p.id === peril.id)}
      />
    ));
};

export class PerilsCategory extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showCategory: false,
    };
  }

  render() {
    let CollapseButton = this.state.showCategory
      ? DisabledCollapseButton
      : DisabledExpandButton;
    return (
      <StyledCategoryContainer
        activeOpacity={1}
        onPress={() =>
          this.setState({ showCategory: !this.state.showCategory })
        }
      >
        <StyledCategoryHeader>
          <StyledCategoryIcon source={{ uri: this.props.iconUrl }} />
          <StyledCategoryTextAndButton>
            <StyledCategoryTextContainer>
              <StyledRow>
                <StyledHeading>{this.props.title}</StyledHeading>
              </StyledRow>
              <StyledRow>
                <StyledPassiveText>
                  {this.props.description}
                  {this.state.showCategory && ':'}
                </StyledPassiveText>
              </StyledRow>
            </StyledCategoryTextContainer>
            <StyledExpandButton>
              <CollapseButton size="mediumBig" />
            </StyledExpandButton>
          </StyledCategoryTextAndButton>
        </StyledCategoryHeader>
        {this.state.showCategory && (
          <ExpandedPerilsCategory>
            {coveredPerils(this.props.perils, this.props.title)}
          </ExpandedPerilsCategory>
        )}
      </StyledCategoryContainer>
    );
  }
}

export const ExpandedPerilsCategory = (props) => (
  <View>
    <StyledPerilsContainer>
      <StyledPerilsRow>{props.children}</StyledPerilsRow>
    </StyledPerilsContainer>
    <StyledPerilsHelpText>
      <StyledSmallPassiveText>
        Klicka på ikonerna för mer info
      </StyledSmallPassiveText>
    </StyledPerilsHelpText>
  </View>
);
