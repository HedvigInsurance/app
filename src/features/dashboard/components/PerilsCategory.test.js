import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { PerilsCategory } from './PerilsCategory';

describe('<PerilsCategory />', () => {
  it('Should render correctly', () => {
    const peril = {
      id: 'ME.LEGAL',
      title: 'Juridisk tvist',
    };
    const perils = [peril];
    const component = shallow(
      <PerilsCategory
        title={'Jag och min familj'}
        description={'Försäkras för'}
        perils={perils}
        iconUrl={''}
        key={0}
        navigation={null}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
