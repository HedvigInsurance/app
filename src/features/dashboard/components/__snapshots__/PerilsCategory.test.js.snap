// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PerilsCategory /> Should render correctly 1`] = `
<StyledCategoryContainer
  activeOpacity={1}
  onPress={[Function]}
>
  <StyledCategoryHeader>
    <StyledCategoryIcon
      source={
        Object {
          "uri": "",
        }
      }
    />
    <StyledCategoryTextAndButton>
      <StyledCategoryTextContainer>
        <Styled(Component)>
          <Styled(Text)>
            Jag och min familj
          </Styled(Text)>
        </Styled(Component)>
        <Styled(Component)>
          <Styled(Text)>
            Försäkras för
          </Styled(Text)>
        </Styled(Component)>
      </StyledCategoryTextContainer>
      <StyledExpandButton>
        <DisabledExpandButton
          size="mediumBig"
        />
      </StyledExpandButton>
    </StyledCategoryTextAndButton>
  </StyledCategoryHeader>
</StyledCategoryContainer>
`;
